src/Administration/Resources/app/administration/src/module/sw-import-export/component/sw-import-export-entity-path-select/index.js

```diff
@@ -422,6 +414,59 @@ export default {
     },
 
     methods: {
+        getDefinition(isCustomField) {
+            if (isCustomField) {
+                return {
+                    properties: this.getCustomFields(this.currentEntity || this.entityType),
+                };
+            }
+
+            // if entity is read or write protected, do not provide any property
+            const definition = Shopware.EntityDefinition.get(this.currentEntity);
+            if (definition.readProtected || definition.writeProtected) {
+                return {
+                    properties: {},
+                };
+            }
+
+            // Filter out properties which are not AdminApiSourceAware
+            const properties = definition.filterProperties((property) => {
+                return this.propertyFilter(property);
+            });
+
+            return {
+                properties: properties,
+            };
+        },
+
+        propertyFilter(property) {
+            const readProtectedLength = property?.flags?.read_protected?.length || 0;
+
+            // if associated entity is read or write protected, do not provide as property
+            if (property.type === 'association') {
+                const subEntity = Shopware.EntityDefinition.get(property.entity);
+                if (subEntity && (subEntity.readProtected || subEntity.writeProtected)) {
+                    return false;
+                }
+            }
+
+            if (readProtectedLength === 0) {
+                return false;
+            }
+
+            const awareKey = 'Shopware\\Core\\Framework\\Api\\Context\\AdminApiSource';
+            if (!property?.flags?.read_protected[0]?.includes(awareKey)) {
+                return false;
+            }
```

Comment:
maybe it makes sense to not make it dependent to the complete namespace

```suggestion
            if (!property?.flags?.read_protected?.[0]?.some((key) => key.split('\\').pop() === 'AdminApiSource')) {
                return false;
            }
```

Reply:
Inside the suggest `.some` lambda, you could also use [endsWith](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith) instead of the string splitting and popping the end of